client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -150,14 +149,25 @@ def process(self, instance):
         self.log.debug("Extracted instance '%s' to: %s",
                        instance.name, json_representation)
 
-    def create_transformation_matrix(self, local_matrix, local_rotation):
+    def create_transformation_matrix(self, local_matrix, local_rotation, product_type, extension="abc"):
         matrix = om.MMatrix(local_matrix)
-        matrix = self.convert_transformation_matrix(matrix, local_rotation)
+        if extension == "fbx":
+            matrix = self.convert_transformation_matrix(matrix, local_rotation)
+        elif extension == "abc":
+            matrix = self.convert_abc_transformation_matrix(matrix, local_rotation)
+        elif extension == "ma":
+            if product_type == "rig":
+                matrix = self.convert_transformation_matrix(matrix, local_rotation)
+            else:
+                matrix = self.convert_abc_transformation_matrix(matrix, local_rotation)
+        else:
+            raise RuntimeError(
+                f"No convert transformation matrix found for the loaded asset in {extension}")
         t_matrix = convert_matrix_to_4x4_list(matrix)
         return t_matrix
 
-    def convert_transformation_matrix(self, transform_mm: om.MMatrix, rotation: list) -> om.MMatrix:
-        """Convert matrix to list of transformation matrix for Unreal Engine import.
+    def convert_abc_transformation_matrix(self, transform_mm: om.MMatrix, rotation: list) -> om.MMatrix:
```

Comment:
This method and `convert_transformation_matrix` appear to be the exact same implementation? They don't differ, or am I blind? :)

And if they _were_ to differ - why do they need to?

Reply:
I know you would ask this - There are some extreme cases of the orientation in terms of alembic import not resolved yet so that's why I left the function like this. If we can any idea to resolve this, and we can go back just edit the function.